src/attr/_make.py

```diff
@@ -368,19 +375,22 @@ def _transform_attrs(cls, these, auto_attribs):
 
     AttrsClass = _make_attr_tuple_class(cls.__name__, attr_names)
 
-    attrs = AttrsClass(
-        super_attrs
-        + [
```

Comment:
OK this one has me my head scratching…did we basically create own_attrs twice until now and nobody noticed? I’m really surprised everything still works. 😬

Reply:
Yes, it took me a few minutes of diffing to convince myself that was the case. 

I *think* this was a happy byproduct of the "frozen derived object" philosophy used in the `Attribute` object, the counting attribute collection was traversed twice, each time producing an effectively identical set of `Attribute` instances.  The [first collection at L343-L348](https://github.com/uw-ipd/attrs/blob/d45b4c3514a326a515b1b854e254606a35a09ff0/src/attr/_make.py#L343-L348) is identical to the [second at L373-378](https://github.com/uw-ipd/attrs/blob/d45b4c3514a326a515b1b854e254606a35a09ff0/src/attr/_make.py#L373-L378)

This fact never "leaked" out of this function to cause issues because: (a) the `own_attrs` collection was only ever used to gather attribute names & the second initialization was immediately passed into the `AttrsClass` container and (b) the factory function [`from_counting_attr`](https://github.com/uw-ipd/attrs/blob/4f92e8377b1a8127392d3d1c84bab15c79f6e184/src/attr/_make.py#L1582) is pure.

To be pedantic, this change requires eliminating this double init because `kw_only` requires evolving the `own_attrs` and `super_attrs` collections to match the class-level configuration and we need to track the evolved collection instance.